test/export/test_export.py

```diff
@@ -995,6 +995,141 @@ def forward(self, x):
         ep_model = export(model, (x,), strict=False).module()
         self.assertTrue(torch.allclose(model(x), ep_model(x)))
 
+    def test_real_tensor_size_mismatch(self):
+        from torch._subclasses.fake_tensor import FakeRealKernelMismatchException
+
+        with torch.library._scoped_library("mylib", "FRAGMENT") as lib:
+            torch.library.define(
+                "mylib::foo",
+                "(Tensor a, Tensor b) -> (Tensor)",
+                tags=torch.Tag.pt2_compliant_tag,
+                lib=lib,
+            )
+
+            class M(torch.nn.Module):
+                def forward(self, a, b):
+                    return torch.ops.mylib.foo(a, b)
+
+            @torch.library.impl("mylib::foo", "cpu", lib=lib)
+            def foo_impl(a, b):
+                return a + b
+
+            @torch.library.register_fake("mylib::foo", lib=lib)
+            def foo_fake_impl(a, b):
+                m, n = a.shape
+                return torch.empty(n, m)  # incorrectly permute
```

Comment:
nit: use torch.library.custom_op unless you're explicitly testing the low-level torch.library APIs